### PR TITLE
sql: update `explain` results

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? (Required)

- Update statement of EXPLAIN with tidb-in-action.
- Update explain result with current result.

**(Note that this PR does not translate `如果读者想要详细了解各算子的内部实现，可以参见[源码阅读系列](https://pingcap.com/blog-cn/#TiDB-源码阅读)。` and `* [TiDB in Action](https://book.tidb.io/session3/chapter1/sql-execution-plan.html)` since the English version does not exist till now. )**

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2548
- Other reference link(s):

### CODE REVIEW

File: sql-statements/sql-statement-explain.md [hunk 1]
```diff
@@ -24,111 +26,196 @@ The statements `DESC` and `DESCRIBE` are aliases of this statement. The alternat
...
+| estRows       | Number of rows that the operator is expected to output. Data is based on statistical information and the evaluation of the operator's logic execution. `estRows` is called `count` in the earlier versions of TiDB 4.0. |
+| task          | The type of task the operator belongs to. Currently, the execution plans are divided into two tasks: **root** task, which is executed on tidb-server, and **cop** task, which is performed in parallel on TiKV or TiFlash. The topology of the execution plan at the task level is that a root task followed by many cop tasks. The root task uses the output of cop tasks as input. The cop tasks refers to tasks that TiDB pushes down to TiKV or TiFlash. Each cop task is distributed in the TiKV cluster or the TiFlash cluster, and is executed by multiple processes. |
+| access object | Data item information accessed by the operator. The information includes `table`, `partition`, and `index` (if any). Only operators who directly access the data have access to such information. |
+| operator info | Other information about the operator. `operator info` of each operator is different, you can refer to the following examples. |
```
Comment (TomShawn) [inline sql-statements/sql-statement-explain.md:39]:
```suggestion
| operator info | Other information about the operator. `operator info` of each operator is different. You can refer to the following examples. |
```

File: sql-statements/sql-statement-explain.md [hunk 2]
```diff
@@ -150,6 +237,13 @@ If the `dot` program is not installed on your computer, copy the result to [this
 * TiDB does not support the `EXPLAIN FORMAT=JSON` as in MySQL.
 * TiDB does not currently support `EXPLAIN` for insert statements.
 
+## `EXPLAIN FOR CONNECTION`
+
+`EXPLAIN FOR CONNECTION` is used to get the execution plan of the last executed query in a connection. The output format is the same as that of `EXPLAIN`. However, the implementation of `EXPLAIN FOR CONNECTION` in TiDB is different from that in MySQL. The following lists their differences apart from the output format:
```
Comment (TomShawn) [inline sql-statements/sql-statement-explain.md:242]:
```suggestion
`EXPLAIN FOR CONNECTION` is used to get the execution plan of the last executed query in a connection. The output format is the same as that of `EXPLAIN`. However, the implementation of `EXPLAIN FOR CONNECTION` in TiDB is different from that in MySQL. Their differences (apart from the output format) are listed as follows:
```

File: sql-statements/sql-statement-explain.md [hunk 3]
```diff
@@ -150,6 +237,13 @@ If the `dot` program is not installed on your computer, copy the result to [this
...
+
+- MySQL returns the **executing** query plan, while TiDB returns the **last executed** query plan.
```
Comment (TomShawn) [inline sql-statements/sql-statement-explain.md:244]:
```suggestion
- MySQL returns the query plan that is **being executing**, while TiDB returns the **last executed** query plan.
```

File: sql-statements/sql-statement-explain.md [hunk 4]
```diff
@@ -24,111 +26,196 @@ The statements `DESC` and `DESCRIBE` are aliases of this statement. The alternat
...
 ![ExplainableStmt](/media/sqlgram/ExplainableStmt.png)
 
+## `EXPLAIN` output format
+
+Currently, `EXPLAIN` in TiDB outputs 5 columns: `id`, `estRows`, `task`, `access object`, `operator info`. Each operator in the execution plan is described by these attributes, with each row in the `EXPLAIN` output describing an operator. The description of each attribute is as follows:
+
+| Attribute name          | Description |
+|:----------------|:----------------------------------------------------------------------------------------------------------|
+| id            | The operator ID is the unique identifier of the operator in the entire execution plan. In TiDB 2.1, the ID is formatted to display the tree structure of the operator. Data flows from the child node to the parent node. One and only one parent node for each operator. |
+| estRows       | Number of rows that the operator is expected to output. Data is based on statistical information and the evaluation of the operator's logic execution. `estRows` is called `count` in the earlier versions of TiDB 4.0. |
```
Comment (qw4990) [inline sql-statements/sql-statement-explain.md:36]: `基于统计信息以及算子的执行逻辑估算而来` means `it is estimated according to statistical information and the operator's logic`.

File: sql-statements/sql-statement-explain.md [hunk 5]
```diff
@@ -24,111 +26,196 @@ The statements `DESC` and `DESCRIBE` are aliases of this statement. The alternat
...
+| task          | The type of task the operator belongs to. Currently, the execution plans are divided into two tasks: **root** task, which is executed on tidb-server, and **cop** task, which is performed in parallel on TiKV or TiFlash. The topology of the execution plan at the task level is that a root task followed by many cop tasks. The root task uses the output of cop tasks as input. The cop tasks refer to tasks that TiDB pushes down to TiKV or TiFlash. Each cop task is distributed in the TiKV cluster or the TiFlash cluster, and is executed by multiple processes. |
+| access object | Data item information accessed by the operator. The information includes `table`, `partition`, and `index` (if any). Only operators that directly access the data have access to such information. |
```
Comment (qw4990) [inline sql-statements/sql-statement-explain.md:38]: `Only operators that directly access the data have access to such information.` -> `Only operators that directly access the data have such information.`?

File: query-execution-plan.md [hunk 6]
```diff
@@ -8,156 +8,471 @@ aliases: ['/docs/dev/query-execution-plan/','/docs/dev/reference/performance/und
...
+## `EXPLAIN` overview
 
 The result of the `EXPLAIN` statement provides information about how TiDB executes SQL queries:
 
-- `EXPLAIN` works together with `SELECT`, `DELETE`, `INSERT`, `REPLACE`, and `UPDATE`.
-- When you run the `EXPLAIN` statement, TiDB returns the final physical execution plan which is optimized by the SQL statement of `EXPLAIN`. In other words, `EXPLAIN` displays the complete information about how TiDB executes the SQL statement, such as in which order, how tables are joined, and what the expression tree looks like. For more information, see [`EXPLAIN` output format](#explain-output-format).
-- TiDB also supports `EXPLAIN [options] FOR CONNECTION connection_id`, but with minor differences from MySQL. See [`EXPLAIN FOR CONNECTION`](#explain-for-connection).
+- `EXPLAIN` works together with statements such as `SELECT` and `DELETE`.
+- When you run the `EXPLAIN` statement, TiDB returns the final physical execution plan which is optimized by the SQL statement of `EXPLAIN`. In other words, `EXPLAIN` displays the complete information about how TiDB executes the SQL statement, such as in which order, how tables are joined, and what the expression tree looks like.
```
Comment (qw4990) [inline query-execution-plan.md:16]: which is optimized by the SQL statement of `EXPLAIN` -> which is optimized by the optimizer

File: sql-statements/sql-statement-explain.md (hunk 4 above)
Comment (qw4990) [inline sql-statements/sql-statement-explain.md:36]: `Data is based on statistical information and the evaluation of the operator's logic execution.` -> `It is estimated according to statistical information and the operator's logic`?

File: query-execution-plan.md [hunk 7]
```diff
@@ -8,156 +8,471 @@ aliases: ['/docs/dev/query-execution-plan/','/docs/dev/reference/performance/und
...
+### Read the execution plan of table scans
 
-## `EXPLAIN ANALYZE` output format
+The operators that perform table scans (of the disk or the TiKV Block Cache) are listed as follows:
 
-As an extension to `EXPLAIN`, `EXPLAIN ANALYZE` will execute the query and provide additional execution statistics in the `execution info` column as follows:
+- **TableFullScan**: Full table scan.
+- **TableRangeScan**: Table scans with the specified range.
+- **TableRowIDScan**: Accurately scans the table data based on the RowID passed down from the upper layer.
+- **IndexFullScan**: Another type of full table scan, except that the index data is scanned, rather than the table data.
```
Comment (qw4990) [inline query-execution-plan.md:114]: It's better to add quotes here, `Another type of "full table scan"`.

File: sql-statements/sql-statement-explain.md (hunk 5 above)
Comment (toutdesuite) [reply sql-statements/sql-statement-explain.md]: All updated in 1c79c71c1c013c0972e6183225b5786c306e9a96

**Pull request merged**